### PR TITLE
Item dispenser fixes #10 and #11

### LINKED ISSUE #10: Arrows should be refilled if the player already has some
If the player has less than three arrows in their inventory and an arrow item drop occurs, they should be refilled to a set of three arrows as opposed to not receiving anything,

### PR DESCRIPTION
**Pull request by Simetraa**

### CODE REVIEW

File: plugin/src/main/java/org/astropeci/omw/item/PeriodicItemDispenser.java [hunk 1]
```diff
@@ -107,6 +107,14 @@ private void giveItemsPeriodically() {
 
             if (!inventory.contains(item.getType()) && inventory.getItemInOffHand().getType() != item.getType()) {
                 inventory.addItem(item);
+            } else if(item.getType() == Material.ARROW) {
```
Comment (LlewVallis) [inline plugin/src/main/java/org/astropeci/omw/item/PeriodicItemDispenser.java]:
```suggestion
            } else if (item.getType() == Material.ARROW) {
```
Slight formatting typo, see the main comment for other issues.

**Review by LlewVallis — CHANGES_REQUESTED** [review]
Few issues here:

1. It's possible to have arrow stacks greater than three in size. This looks like it'll truncate stacks down unduly.
2. This doesn't look like it'll send a message to the player if they miss out on arrows due to having too many. Also I believe the display name for an arrow stack is "Arrow" (the message should say "Arrows") so that would cause issues even if there were arrow messages.

**Comment by Simetraa** [comment]
> This doesn't look like it'll send a message to the player if they miss out on arrows due to having too many. Also I believe the display name for an arrow stack is "Arrow" (the message should say "Arrows") so that would cause issues even if there were arrow messages.

Should it send the same message (" already obtained!") if topping up arrows? I checked on CK and it does not send any message.

**Comment by LlewVallis** [comment]
> > This doesn't look like it'll send a message to the player if they miss out on arrows due to having too many. Also I believe the display name for an arrow stack is "Arrow" (the message should say "Arrows") so that would cause issues even if there were arrow messages.
> 
> Should it send the same message (" already obtained!") if topping up arrows? I checked on CK and it does not send any message.

Yes and no, it should only send the "already obtained" message for arrows if the player already has three or more in their inventory (and hence doesn't receive any extras).

**Review by LlewVallis — COMMENTED** [review]
Looks good at this point! Thanks :) I'll fully test and merge the changes when I have a chance.

**Comment by LlewVallis** [comment]
I've validated that everything is correct and working, and hence have merged it into master. Thanks for the contribution!

**Pull request merged**